mono/mini/aot-compiler.c

```diff
@@ -9052,7 +9052,7 @@ append_mangled_method (GString *s, MonoMethod *method)
 		g_assert (imethod->context.class_inst != NULL || imethod->context.method_inst != NULL);
 
 		append_mangled_context (s, &imethod->context);
-		g_string_append_printf (s, "_declared_by_");
+		g_string_append_printf (s, "_declared_by_%s_", imethod->declaring->klass->image->assembly->aname.name);
```

Comment:
can there be issues if the assembly name is containing dots or some other characters ?